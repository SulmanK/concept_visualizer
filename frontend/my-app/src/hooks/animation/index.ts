export { default as useAnimatedMount } from "./useAnimatedMount";
export { default as useAnimatedValue } from "./useAnimatedValue";
export { default as usePrefersReducedMotion } from "./usePrefersReducedMotion";

// Re-export types
export type { AnimationState, AnimatedMountOptions } from "./useAnimatedMount";
export type { AnimatedValueOptions } from "./useAnimatedValue";
