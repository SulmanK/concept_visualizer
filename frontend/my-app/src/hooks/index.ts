// API Hooks
export { useConceptDetail, useRecentConcepts } from "./useConceptQueries";
export {
  useGenerateConceptMutation,
  useRefineConceptMutation,
} from "./useConceptMutations";
export {
  useRateLimitsQuery,
  useOptimisticRateLimitUpdate,
} from "./useRateLimitsQuery";
export { useExportImageMutation, downloadBlob } from "./useExportImageMutation";
export { useTaskStatusQuery, useTaskCancelMutation } from "./useTaskQueries";
export { useTaskSubscription } from "./useTaskSubscription";
export { useSessionSyncMutation } from "./useSessionQuery";
export { useConfigQuery } from "./useConfigQuery";

// Auth Hooks
export {
  useAuth,
  useAuthUser,
  useUserId,
  useIsAnonymous,
  useAuthIsLoading,
} from "./useAuth";

// Rate Limit Hooks
export {
  useRateLimitsData,
  useRateLimitsLoading,
  useRateLimitsError,
  useRateLimitsRefetch,
  useRateLimitsDecrement,
  useRateLimitContext,
  useDecrementRateLimit,
} from "./useRateLimits";

// Task Hooks
export {
  useTaskContext,
  useHasActiveTask,
  useIsTaskProcessing,
  useIsTaskPending,
  useIsTaskCompleted,
  useIsTaskFailed,
  useLatestResultId,
  useTaskInitiating,
  useActiveTaskId,
  useOnTaskCleared,
  useActiveTaskData,
  useClearActiveTask,
} from "./useTask";

// Error and Loading Hooks
export { useToast } from "./useToast";
export { ToastProvider } from "./toast/ToastContext";
export { useErrorHandling } from "./useErrorHandling";
export { useNetworkStatus } from "./useNetworkStatus";

// Animation Hooks
export {
  useAnimatedMount,
  useAnimatedValue,
  usePrefersReducedMotion,
} from "./animation";

// Animation Types (re-exported)
export type {
  AnimationState,
  AnimatedMountOptions,
  AnimatedValueOptions,
} from "./animation";
