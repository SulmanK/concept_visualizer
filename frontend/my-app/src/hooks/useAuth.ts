import { useContextSelector } from "use-context-selector";
import { AuthContext } from "../contexts/AuthContext";

/**
 * Hook to access the full auth context value
 * @returns The complete auth context
 * @throws Error if used outside of AuthProvider
 */
export const useAuth = () => {
  const context = useContextSelector(AuthContext, (state) => state);
  if (context === undefined) {
    throw new Error("useAuth must be used within an AuthProvider");
  }
  return context;
};

/**
 * Hook to access just the user object from auth context
 * @returns The current user or null
 * @throws Error if used outside of AuthProvider
 */
export const useAuthUser = () => {
  const context = useContextSelector(AuthContext, (state) => state?.user);
  if (context === undefined) {
    throw new Error("useAuthUser must be used within an AuthProvider");
  }
  return context;
};

/**
 * Hook to access just the user ID from auth context
 * @returns The current user ID or null
 */
export const useUserId = () => {
  const context = useContextSelector(AuthContext, (state) => state?.user?.id);
  if (context === undefined && process.env.NODE_ENV !== "production") {
    console.warn("useUserId called outside AuthProvider or no user exists");
  }
  return context || null;
};

/**
 * Hook to access the anonymous status from auth context
 * @returns Boolean indicating if the current user is anonymous
 * @throws Error if used outside of AuthProvider
 */
export const useIsAnonymous = () => {
  const context = useContextSelector(
    AuthContext,
    (state) => state?.isAnonymous,
  );
  if (context === undefined) {
    throw new Error("useIsAnonymous must be used within an AuthProvider");
  }
  return context;
};

/**
 * Hook to access the loading state from auth context
 * @returns Boolean indicating if auth is currently loading
 * @throws Error if used outside of AuthProvider
 */
export const useAuthIsLoading = () => {
  const context = useContextSelector(AuthContext, (state) => state?.isLoading);
  if (context === undefined) {
    throw new Error("useAuthIsLoading must be used within an AuthProvider");
  }
  return context;
};
