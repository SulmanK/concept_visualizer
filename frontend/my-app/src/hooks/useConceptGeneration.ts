/**
 * Hook for generating visual concepts via the API.
 */

import { useState, useCallback } from 'react';
import { useApi } from './useApi';
import { 
  GenerationResponse,
  PromptRequest,
  FormStatus
} from '../types';
import { useErrorHandling } from './useErrorHandling';
import { useRateLimitContext } from '../contexts/RateLimitContext';
import { createAsyncErrorHandler } from '../utils/errorUtils';
import { useQueryClient } from '@tanstack/react-query';
import { useAuth } from '../contexts/AuthContext';

export interface ConceptGenerationState {
  status: FormStatus;
  result: GenerationResponse | null;
  error: string | null;
}

/**
 * Hook for generating visual concepts based on descriptions
 */
export function useConceptGeneration() {
  const { post, loading, error, clearError } = useApi();
  const { user } = useAuth();
  const queryClient = useQueryClient();
  const errorHandler = useErrorHandling();
  const { decrementLimit } = useRateLimitContext();
  
  // Create a standardized async error handler
  const handleAsync = createAsyncErrorHandler(errorHandler, {
    defaultErrorMessage: 'Failed to generate concept',
    showToast: true
  });
  
  const [generationState, setGenerationState] = useState<ConceptGenerationState>({
    status: 'idle',
    result: null,
    error: null,
  });

  /**
   * Generate a new concept based on provided descriptions
   */
  const generateConcept = useCallback(async (
    logoDescription: string,
    themeDescription: string
  ) => {
    // Validate inputs
    if (!logoDescription.trim() || !themeDescription.trim()) {
      setGenerationState({
        status: 'error',
        result: null,
        error: 'Please provide both logo and theme descriptions',
      });
      return;
    }

    // Set submitting state
    setGenerationState({
      status: 'submitting',
      result: null,
      error: null,
    });

    // Apply optimistic update BEFORE making the API call
    // This immediately decrements the count in the UI
    decrementLimit('generate_concept');

    // Use our standardized error handler to wrap the API call
    const result = await handleAsync(async () => {
      const request: PromptRequest = {
        logo_description: logoDescription,
        theme_description: themeDescription,
      };

      // Use the generate-with-palettes endpoint to get variations
      const response = await post<GenerationResponse>('/concepts/generate-with-palettes', request);

      if (response.error) {
        throw new Error(response.error.message);
      }

      if (!response.data) {
        throw new Error('No data returned from API');
      }

      // Make sure the variations array is included in the result
      return {
        ...response.data,
        variations: response.data.variations || []
      };
    }, 'concept generation');

    if (result) {
      console.log('Generated concept with variations:', result);
      
      setGenerationState({
        status: 'success',
        result,
        error: null,
      });
      
      // REPLACING EVENT EMISSION WITH DIRECT CACHE INVALIDATION
      // Instead of using events, directly invalidate the concepts cache
      console.log('Directly invalidating concepts query cache');
      
      // Invalidate the recent concepts query
      queryClient.invalidateQueries({ 
        queryKey: ['concepts', 'recent', user?.id] 
      });
      
      // Also invalidate any specific concept query if needed
      if (result.id) {
        queryClient.invalidateQueries({ 
          queryKey: ['concepts', 'detail', result.id] 
        });
      }
      
      // Also invalidate any other related queries
      queryClient.invalidateQueries({
        queryKey: ['concepts'],
        exact: false
      });
    } else {
      // If result is undefined, an error occurred and was handled by handleAsync
      setGenerationState({
        status: 'error',
        result: null,
        error: errorHandler.error?.message || 'Failed to generate concept',
      });
    }
  }, [post, queryClient, user?.id, decrementLimit, handleAsync, errorHandler]);

  /**
   * Reset the generation state
   */
  const resetGeneration = useCallback(() => {
    setGenerationState({
      status: 'idle',
      result: null,
      error: null,
    });
    clearError();
    errorHandler.clearError();
  }, [clearError, errorHandler]);

  return {
    generateConcept,
    resetGeneration,
    ...generationState,
    isLoading: generationState.status === 'submitting',
  };
} 