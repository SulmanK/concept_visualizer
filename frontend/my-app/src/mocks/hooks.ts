import React from "react";
import { vi } from "vitest";

// =============================================================================
// Mock Auth Context
// =============================================================================

// Create a context for our mock auth
export const mockAuthContext = React.createContext({
  user: { id: "test-user-id", email: "test@example.com" },
  session: { user: { id: "test-user-id", email: "test@example.com" } },
  isLoading: false,
  isAuthenticated: true,
  isAnonymous: false,
  signOut: vi.fn(),
  linkEmail: vi.fn(),
  refreshSession: vi.fn(),
  error: null,
});

// Mock auth context values
export const mockAuthContextValue = {
  user: { id: "test-user-id", email: "test@example.com" },
  session: { user: { id: "test-user-id", email: "test@example.com" } },
  isLoading: false,
  isAuthenticated: true,
  isAnonymous: false,
  signOut: vi.fn(),
  linkEmail: vi.fn(),
  refreshSession: vi.fn(),
  error: null,
};

// Create the auth hooks that match the ones in the real AuthContext
export const useAuth = () => React.useContext(mockAuthContext);
export const useAuthUser = () => React.useContext(mockAuthContext).user;
export const useUserId = () => React.useContext(mockAuthContext).user?.id;
export const useIsAnonymous = () =>
  React.useContext(mockAuthContext).isAnonymous;
export const useAuthIsLoading = () =>
  React.useContext(mockAuthContext).isLoading;

// =============================================================================
// Mock Toast Context
// =============================================================================

// Create a mock Toast context
export const mockToastContext = React.createContext({
  showToast: vi.fn(),
  showSuccess: vi.fn(),
  showError: vi.fn(),
  showWarning: vi.fn(),
  showInfo: vi.fn(),
  dismissToast: vi.fn(),
  dismissAll: vi.fn(),
});

// Hook for accessing toast context
export const useToast = () => React.useContext(mockToastContext);

// Mock toast context values
export const mockToastContextValue = {
  showToast: vi.fn(),
  showSuccess: vi.fn(),
  showError: vi.fn(),
  showWarning: vi.fn(),
  showInfo: vi.fn(),
  dismissToast: vi.fn(),
  dismissAll: vi.fn(),
};
