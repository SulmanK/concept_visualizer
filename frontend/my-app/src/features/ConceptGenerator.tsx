import React, { useState } from 'react';
import { ConceptForm } from '../components/concept/ConceptForm';
import { ConceptResult } from '../components/concept/ConceptResult';
import { GenerationResponse } from '../types';
import { useConceptGeneration } from '../hooks/useConceptGeneration';

/**
 * Feature component for generating new concepts
 */
export const ConceptGenerator: React.FC = () => {
  const { 
    generateConcept, 
    resetGeneration, 
    status, 
    result, 
    error,
    isLoading
  } = useConceptGeneration();
  
  const [selectedColor, setSelectedColor] = useState<string | null>(null);
  
  const handleGenerateConcept = (logoDescription: string, themeDescription: string) => {
    generateConcept(logoDescription, themeDescription);
    setSelectedColor(null);
  };
  
  const handleReset = () => {
    resetGeneration();
    setSelectedColor(null);
  };
  
  const handleColorSelect = (color: string) => {
    setSelectedColor(color);
    
    // Copy color to clipboard
    navigator.clipboard.writeText(color)
      .then(() => {
        // Could show a toast notification here
        console.log(`Copied ${color} to clipboard`);
      })
      .catch(err => {
        console.error('Could not copy text: ', err);
      });
  };
  
  return (
    <div className="space-y-8">
      <div className="text-center mb-8">
        <h1 className="text-3xl font-bold text-dark-900 mb-2">
          Create Visual Concepts
        </h1>
        <p className="text-dark-600 max-w-2xl mx-auto">
          Describe your logo and theme to generate visual concepts. Our AI will create a logo design and color palette based on your descriptions.
        </p>
      </div>
      
      <ConceptForm
        onSubmit={handleGenerateConcept}
        status={status}
        error={error}
        onReset={handleReset}
      />
      
      {status === 'success' && result && (
        <div className="mt-8 pt-8 border-t border-dark-200">
          <ConceptResult
            concept={result}
            onRefineRequest={() => {
              // This would navigate to the refinement page with the current concept
              console.log('Navigate to refinement with:', result.generationId);
            }}
            onColorSelect={(color) => handleColorSelect(color)}
          />
          
          {selectedColor && (
            <div className="mt-4 text-center text-sm text-dark-600">
              <span className="bg-dark-100 px-2 py-1 rounded font-mono">
                {selectedColor}
              </span>
              <span className="ml-2">
                copied to clipboard
              </span>
            </div>
          )}
        </div>
      )}
    </div>
  );
}; 