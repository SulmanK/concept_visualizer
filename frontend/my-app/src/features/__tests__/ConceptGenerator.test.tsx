import React from 'react';
import { render, screen, fireEvent, within } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ConceptGenerator } from '../ConceptGenerator';
import * as useConceptGenerationModule from '../../hooks/useConceptGeneration';

// Mock console.log
const originalConsoleLog = console.log;
console.log = jest.fn();

// Clean up after tests
afterAll(() => {
  console.log = originalConsoleLog;
});

// Mock the useConceptGeneration hook
jest.mock('../../hooks/useConceptGeneration');

// Mock the useNavigate hook from react-router-dom
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

// Helper function to render with router
const renderWithRouter = (ui: React.ReactElement) => {
  return render(
    <BrowserRouter>
      {ui}
    </BrowserRouter>
  );
};

describe('ConceptGenerator Component', () => {
  // Mock implementation for useConceptGeneration
  const mockGenerateConcept = jest.fn();
  const mockResetGeneration = jest.fn();
  
  // Set up mock response data with correct structure
  const mockResult = {
    imageUrl: 'https://example.com/image.png',
    generationId: 'concept-123',
    createdAt: '2023-03-01T12:00:00Z',
    colorPalette: {
      primary: '#4F46E5',
      secondary: '#60A5FA',
      accent: '#EEF2FF',
      background: '#FFFFFF',
      text: '#1E293B',
      additionalColors: ['#818CF8', '#6366F1']
    }
  };
  
  beforeEach(() => {
    // Reset all mocks
    jest.clearAllMocks();
    
    // Default mock implementation
    jest.spyOn(useConceptGenerationModule, 'useConceptGeneration').mockImplementation(() => ({
      generateConcept: mockGenerateConcept,
      resetGeneration: mockResetGeneration,
      status: 'idle',
      result: null,
      error: null,
      isLoading: false
    }));
  });
  
  // Basic rendering tests
  test('renders the component in initial state', () => {
    renderWithRouter(<ConceptGenerator />);
    
    // Check for page title
    const title = screen.getByText('Create Visual Concepts');
    expect(title).toBeInTheDocument();
    
    // Check for How It Works section
    const howItWorksTitle = screen.getByText('How It Works');
    expect(howItWorksTitle).toBeInTheDocument();
    
    // Check for sample concepts section
    const recentConcepts = screen.getByText(/Recent Concepts/i);
    expect(recentConcepts).toBeInTheDocument();
  });
  
  // Form submission test
  test('submits the form and calls generateConcept', () => {
    renderWithRouter(<ConceptGenerator />);
    
    // Find the form inputs by their labels
    const logoLabel = screen.getByText('Logo Description');
    const themeLabel = screen.getByText('Theme/Color Scheme Description');
    
    // Find the textareas that are siblings of the labels
    const logoTextarea = logoLabel.parentElement?.querySelector('textarea');
    const themeTextarea = themeLabel.parentElement?.querySelector('textarea');
    
    // Make sure we found the textareas
    expect(logoTextarea).toBeInTheDocument();
    expect(themeTextarea).toBeInTheDocument();
    
    if (logoTextarea && themeTextarea) {
      // Fill out the form
      fireEvent.change(logoTextarea, { target: { value: 'A modern logo' } });
      fireEvent.change(themeTextarea, { target: { value: 'Blue and purple theme' } });
      
      // Find and submit the form
      const submitButton = screen.getByRole('button', { name: /Generate Concept/i });
      fireEvent.click(submitButton);
      
      // Check if the generateConcept function was called with correct args
      expect(mockGenerateConcept).toHaveBeenCalledWith('A modern logo', 'Blue and purple theme');
    }
  });
  
  // Loading state test
  test('displays loading state when generating concept', () => {
    // Mock loading state
    jest.spyOn(useConceptGenerationModule, 'useConceptGeneration').mockImplementation(() => ({
      generateConcept: mockGenerateConcept,
      resetGeneration: mockResetGeneration,
      status: 'submitting',
      result: null,
      error: null,
      isLoading: true
    }));
    
    renderWithRouter(<ConceptGenerator />);
    
    // Check for loading indicator
    const loadingButton = screen.getByText('Generating...');
    expect(loadingButton).toBeInTheDocument();
    expect(loadingButton.closest('button')).toBeDisabled();
  });
  
  // Success state test
  test('displays result when concept generation succeeds', () => {
    // Mock success state
    jest.spyOn(useConceptGenerationModule, 'useConceptGeneration').mockImplementation(() => ({
      generateConcept: mockGenerateConcept,
      resetGeneration: mockResetGeneration,
      status: 'success',
      result: mockResult,
      error: null,
      isLoading: false
    }));
    
    renderWithRouter(<ConceptGenerator />);
    
    // Check for the generated image
    const resultImage = screen.getAllByRole('img').find(img => 
      img.getAttribute('src') === mockResult.imageUrl
    );
    expect(resultImage).toBeInTheDocument();
    
    // Check for primary color display (since the palette names may vary)
    const primaryColorText = screen.getByText('Primary');
    expect(primaryColorText).toBeInTheDocument();
    
    // Check for action buttons - looking for the Refine/Download buttons that are present in the success state
    const refineButton = screen.getByRole('button', { name: /Refine This Concept/i });
    const downloadButton = screen.getByRole('button', { name: /Download Image/i });
    expect(refineButton).toBeInTheDocument();
    expect(downloadButton).toBeInTheDocument();
  });
  
  // Error state test
  test('displays error message when concept generation fails', () => {
    const errorMessage = 'Failed to generate concept';
    
    // Mock error state
    jest.spyOn(useConceptGenerationModule, 'useConceptGeneration').mockImplementation(() => ({
      generateConcept: mockGenerateConcept,
      resetGeneration: mockResetGeneration,
      status: 'error',
      result: null,
      error: errorMessage,
      isLoading: false
    }));
    
    renderWithRouter(<ConceptGenerator />);
    
    // Check for error message
    const errorElement = screen.getByText(errorMessage);
    expect(errorElement).toBeInTheDocument();
    
    // Form should still be accessible
    const submitButton = screen.getByRole('button', { name: /Generate Concept/i });
    expect(submitButton).toBeInTheDocument();
  });
  
  // Reset test - update to match actual implementation
  test('navigates to refinement when Refine This Concept is clicked', () => {
    // Mock success state
    jest.spyOn(useConceptGenerationModule, 'useConceptGeneration').mockImplementation(() => ({
      generateConcept: mockGenerateConcept,
      resetGeneration: mockResetGeneration,
      status: 'success',
      result: mockResult,
      error: null,
      isLoading: false
    }));
    
    renderWithRouter(<ConceptGenerator />);
    
    // Find and click the Refine This Concept button
    const refineButton = screen.getByRole('button', { name: /Refine This Concept/i });
    fireEvent.click(refineButton);
    
    // In the real implementation, this would navigate to the refinement page
    // with the concept ID, not actually call resetGeneration
    expect(console.log).toHaveBeenCalledWith('Navigate to refinement with:', mockResult.generationId);
  });
  
  // Snapshot test
  test('matches snapshot in initial state', () => {
    const { container } = renderWithRouter(<ConceptGenerator />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 