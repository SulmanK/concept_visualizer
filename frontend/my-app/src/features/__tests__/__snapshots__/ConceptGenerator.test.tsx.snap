// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConceptGenerator Component > matches snapshot in initial state 1`] = `
<div
  class="space-y-12"
>
  <div
    class="text-left mb-8"
  >
    <h1
      class="text-4xl font-bold text-indigo-900 mb-4"
    >
      Create Visual Concepts
    </h1>
    <p
      style="color: rgb(99, 102, 241); font-size: 1rem; line-height: 1.5; margin-bottom: 2rem;"
    >
      Describe your logo and theme to generate visual concepts. Our AI will create a logo design and color palette based on your descriptions.
    </p>
  </div>
  <div
    class="mt-8"
    style="background-color: white; border-radius: 0.75rem; box-shadow: 0 10px 25px -5px rgba(0, 0, 0, 0.05), 0 8px 10px -6px rgba(0, 0, 0, 0.02); padding: 2rem; margin-bottom: 3rem;"
  >
    <h2
      class="text-2xl font-bold text-indigo-900 mb-16"
    >
      How It Works
    </h2>
    <div
      style="display: grid; grid-template-columns: repeat(3, 1fr); gap: 1.5rem; margin-bottom: 2rem; margin-top: 2rem; position: relative;"
    >
      <div
        style="position: absolute; top: 30px; left: calc(16.0% + 35px); width: calc(33.3% - 70px); height: 2px; background-color: rgb(99, 102, 241); z-index: 0;"
      >
        <div
          style="position: absolute; right: -8px; top: -4px; width: 0px; height: 0px; border-top: 5px solid transparent; border-bottom: 5px solid transparent; border-left: 8px solid #6366F1;"
        />
      </div>
      <div
        style="position: absolute; top: 30px; left: calc(50% + 35px); width: calc(33.3% - 70px); height: 2px; background-color: rgb(99, 102, 241); z-index: 0;"
      >
        <div
          style="position: absolute; right: -8px; top: -4px; width: 0px; height: 0px; border-top: 5px solid transparent; border-bottom: 5px solid transparent; border-left: 8px solid #6366F1;"
        />
      </div>
      <div
        style="text-align: center; position: relative; z-index: 1;"
      >
        <div
          style="width: 60px; height: 60px; background-color: rgb(238, 242, 255); color: rgb(99, 102, 241); border-radius: 50%; display: flex; align-items: center; justify-content: center; font-size: 20px; font-weight: 600; margin: 0px auto;"
        >
          1
        </div>
        <h3
          style="font-size: 1.125rem; font-weight: 600; color: rgb(30, 41, 59); margin-bottom: 1rem;"
        >
          Describe Your Vision
        </h3>
        <p
          style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem;"
        >
          Provide detailed descriptions of your logo concept and color preferences.
        </p>
      </div>
      <div
        style="text-align: center; position: relative; z-index: 1;"
      >
        <div
          style="width: 60px; height: 60px; background-color: rgb(238, 242, 255); color: rgb(99, 102, 241); border-radius: 50%; display: flex; align-items: center; justify-content: center; font-size: 20px; font-weight: 600; margin: 0px auto;"
        >
          2
        </div>
        <h3
          style="font-size: 1.125rem; font-weight: 600; color: rgb(30, 41, 59); margin-bottom: 1rem;"
        >
          AI Generation
        </h3>
        <p
          style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem;"
        >
          Our AI processes your description and creates unique visual concepts.
        </p>
      </div>
      <div
        style="text-align: center; position: relative; z-index: 1;"
      >
        <div
          style="width: 60px; height: 60px; background-color: rgb(238, 242, 255); color: rgb(99, 102, 241); border-radius: 50%; display: flex; align-items: center; justify-content: center; font-size: 20px; font-weight: 600; margin: 0px auto;"
        >
          3
        </div>
        <h3
          style="font-size: 1.125rem; font-weight: 600; color: rgb(30, 41, 59); margin-bottom: 1rem;"
        >
          Refine & Download
        </h3>
        <p
          style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem;"
        >
          Refine the generated concepts and download your final designs.
        </p>
      </div>
    </div>
    <div
      style="text-align: center; margin-top: 2rem;"
    >
      <button
        style="background-color: rgb(79, 70, 229); color: white; padding: 0.75rem 1.5rem; border-radius: 0.375rem; font-weight: 600; font-size: 0.875rem; cursor: pointer; transition: background-color 0.2s;"
      >
        Get Started
      </button>
    </div>
  </div>
  <div
    id="create-form"
  >
    <div
      style="background-color: white; border-radius: 0.75rem; box-shadow: 0 10px 25px -5px rgba(0, 0, 0, 0.05), 0 8px 10px -6px rgba(0, 0, 0, 0.02); padding: 2rem; margin-bottom: 2rem;"
    >
      <h2
        class="text-xl font-semibold text-indigo-900 mb-6"
      >
        Create New Concept
      </h2>
      <form
        class="space-y-6"
      >
        <div>
          <div
            class="mb-6"
          >
            <label
              style="display: block; font-size: 0.875rem; font-weight: 600; color: rgb(67, 56, 202); margin-bottom: 0.5rem;"
            >
              Logo Description
            </label>
            <textarea
              placeholder="Describe the logo you want to generate..."
              style="width: 100%; padding: 0.75rem; border: 1px solid rgb(224, 231, 255); border-radius: 0.375rem; font-size: 0.875rem; background-color: rgb(245, 247, 255); resize: vertical; min-height: 7rem; outline: none; transition: border-color 0.2s, box-shadow 0.2s;"
            />
            <p
              style="font-size: 0.75rem; color: rgb(99, 102, 241); margin-top: 0.25rem;"
            >
              Be descriptive about style, symbols, and colors you want
            </p>
          </div>
          <div
            class="mb-6"
          >
            <label
              style="display: block; font-size: 0.875rem; font-weight: 600; color: rgb(67, 56, 202); margin-bottom: 0.5rem;"
            >
              Theme/Color Scheme Description
            </label>
            <textarea
              placeholder="Describe the theme or color scheme..."
              style="width: 100%; padding: 0.75rem; border: 1px solid rgb(224, 231, 255); border-radius: 0.375rem; font-size: 0.875rem; background-color: rgb(245, 247, 255); resize: vertical; min-height: 7rem; outline: none; transition: border-color 0.2s, box-shadow 0.2s;"
            />
            <p
              style="font-size: 0.75rem; color: rgb(99, 102, 241); margin-top: 0.25rem;"
            >
              Describe mood, colors, and style of your brand
            </p>
          </div>
        </div>
        <div
          class="flex justify-end"
        >
          <button
            style="background-color: rgb(79, 70, 229); color: white; padding: 0.75rem 1.5rem; border-radius: 0.375rem; font-weight: 600; cursor: pointer; transition: background-color 0.2s; margin-top: 1rem;"
            type="submit"
          >
            Generate Concept
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    class="mt-20"
  >
    <h2
      class="text-2xl font-bold text-indigo-900 mb-10"
    >
      Recent Concepts
    </h2>
    <div
      style="display: grid; grid-template-columns: repeat(3, 1fr); gap: 2rem; margin: 2.5rem 0px 0px 0px; padding: 0px; width: 100%;"
    >
      <div
        style="display: flex; flex-direction: column; background-color: white; border-radius: 0.5rem; overflow: hidden; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1); height: 100%;"
      >
        <div
          style="background: linear-gradient(to right, #4F46E5, #60A5FA); height: 200px; display: flex; align-items: center; justify-content: center;"
        >
          <div
            style="width: 70px; height: 70px; background-color: white; border-radius: 50%; display: flex; align-items: center; justify-content: center; font-weight: bold; font-size: 20px; color: rgb(79, 70, 229);"
          >
            TC
          </div>
        </div>
        <div
          style="padding: 1.25rem 1.5rem; flex-grow: 1;"
        >
          <h3
            style="font-weight: 600; font-size: 1rem; color: rgb(30, 41, 59); margin-bottom: 0.5rem;"
          >
            Tech Company
          </h3>
          <p
            style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem; margin-bottom: 1rem;"
          >
            Modern minimalist tech logo with abstract elements
          </p>
          <div
            style="display: flex; gap: 0.5rem; margin-bottom: 1.5rem;"
          >
            <span
              style="background-color: rgb(79, 70, 229); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#4F46E5"
            />
            <span
              style="background-color: rgb(96, 165, 250); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#60A5FA"
            />
            <span
              style="background-color: rgb(30, 41, 59); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#1E293B"
            />
          </div>
          <div
            style="display: flex; justify-content: space-between; padding-top: 1rem; border-top: 1px solid #E2E8F0;"
          >
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              Edit
            </a>
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              View Details
            </a>
          </div>
        </div>
      </div>
      <div
        style="display: flex; flex-direction: column; background-color: white; border-radius: 0.5rem; overflow: hidden; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1); height: 100%;"
      >
        <div
          style="background: linear-gradient(to right, #7E22CE, #818CF8); height: 200px; display: flex; align-items: center; justify-content: center;"
        >
          <div
            style="width: 70px; height: 70px; background-color: white; border-radius: 50%; display: flex; align-items: center; justify-content: center; font-weight: bold; font-size: 20px; color: rgb(126, 34, 206);"
          >
            FS
          </div>
        </div>
        <div
          style="padding: 1.25rem 1.5rem; flex-grow: 1;"
        >
          <h3
            style="font-weight: 600; font-size: 1rem; color: rgb(30, 41, 59); margin-bottom: 0.5rem;"
          >
            Fashion Studio
          </h3>
          <p
            style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem; margin-bottom: 1rem;"
          >
            Elegant fashion brand with clean typography
          </p>
          <div
            style="display: flex; gap: 0.5rem; margin-bottom: 1.5rem;"
          >
            <span
              style="background-color: rgb(126, 34, 206); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#7E22CE"
            />
            <span
              style="background-color: rgb(129, 140, 248); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#818CF8"
            />
            <span
              style="background-color: rgb(249, 168, 212); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#F9A8D4"
            />
          </div>
          <div
            style="display: flex; justify-content: space-between; padding-top: 1rem; border-top: 1px solid #E2E8F0;"
          >
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              Edit
            </a>
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              View Details
            </a>
          </div>
        </div>
      </div>
      <div
        style="display: flex; flex-direction: column; background-color: white; border-radius: 0.5rem; overflow: hidden; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1); height: 100%;"
      >
        <div
          style="background: linear-gradient(to right, #059669, #60A5FA); height: 200px; display: flex; align-items: center; justify-content: center;"
        >
          <div
            style="width: 70px; height: 70px; background-color: white; border-radius: 50%; display: flex; align-items: center; justify-content: center; font-weight: bold; font-size: 20px; color: rgb(5, 150, 105);"
          >
            EP
          </div>
        </div>
        <div
          style="padding: 1.25rem 1.5rem; flex-grow: 1;"
        >
          <h3
            style="font-weight: 600; font-size: 1rem; color: rgb(30, 41, 59); margin-bottom: 0.5rem;"
          >
            Eco Product
          </h3>
          <p
            style="font-size: 0.875rem; color: rgb(100, 116, 139); line-height: 1.25rem; margin-bottom: 1rem;"
          >
            Sustainable brand with natural elements
          </p>
          <div
            style="display: flex; gap: 0.5rem; margin-bottom: 1.5rem;"
          >
            <span
              style="background-color: rgb(5, 150, 105); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#059669"
            />
            <span
              style="background-color: rgb(96, 165, 250); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#60A5FA"
            />
            <span
              style="background-color: rgb(16, 185, 129); width: 1.5rem; height: 1.5rem; border-radius: 9999px; display: inline-block;"
              title="#10B981"
            />
          </div>
          <div
            style="display: flex; justify-content: space-between; padding-top: 1rem; border-top: 1px solid #E2E8F0;"
          >
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              Edit
            </a>
            <a
              href="#"
              style="color: rgb(99, 102, 241); font-weight: 500; font-size: 0.875rem; text-decoration: none; cursor: pointer; padding: 0.25rem 0px;"
            >
              View Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
